compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -264,7 +264,12 @@ object Completion {
     private def accessibleMembers(site: Type)(using Context): Seq[Symbol] = site match {
       case site: NamedType if site.symbol.is(Package) =>
         // Don't look inside package members -- it's too expensive.
```

Comment:
It looks like we were intentionally not taking package object members into account, @odersky do you remember the context of this comment? (added in https://github.com/lampepfl/dotty/commit/518e9fd7465f8a13b641f1ce3298efc80ca259c7)

Reply:
This comment first appeared in this commit https://github.com/lampepfl/dotty/commit/518e9fd if that might help

Reply:
I would like to get this PR finally merged but I'm not sure what I should do to push it forward. If the comment is still valid, I can revert some of the changes and leave only the fix for the regression but from user's perspective completion not working for symbols like `List` or `Iterator` might be quite confusing and irritating.

Reply:
Just asked Martin IRL about this and he thinks it's fine to make this change. So LGTM from me, but please remove that comment since it's no longer accurate.

Reply:
done